loki/frontend/tests/test_frontends.py

```diff
@@ -1015,3 +1015,38 @@ def test_frontend_derived_type_imports(tmp_path, frontend):
     assert isinstance(assigns[1].lhs, sym.Array)
     assert assigns[1].lhs.type.dtype == BasicType.REAL
     assert assigns[1].lhs.type.shape == (':',)
+
+
+@pytest.mark.skipif(not HAVE_FP, reason="Assumed size declarations only supported for FP")
+def test_assumed_size_declarations():
+    """
+    Test if assumed size declarations are correctly parsed.
+    """
+
+    fcode = """
+subroutine kernel(a, b, c)
+  implicit none
+  real, intent(in) :: a(*)
+  real, intent(in) :: b(8,*)
+  real, intent(in) :: c(8,0:*)
```

Comment:
Would you mind adding a `d(2,4,3:*)` to demonstrate that non-zero lower bounds and multiple preceeding explicit-shape-specs are working as well? It looks to me like the implementation would handle this correctly but worth double-checking.